src/sysc/kernel/sc_name_gen.cpp

```diff
@@ -73,6 +73,7 @@ sc_name_gen::gen_unique_name( const char* basename_, bool preserve_first )
             sstr << basename_ << "_" << *c;
             sstr.str().swap( m_unique_name );
         }
+        delete c;
```

Comment:
If I understand the code correctly, `m_unique_name_map` takes over ownership of `c`. Therefore, this is not the right place for deleting it.